triggers/service/src/main/scala/com/digitalasset/daml/lf/engine/trigger/ServiceMain.scala

```diff
@@ -212,30 +202,83 @@ object Server {
-                case Left(err) =>
-                  complete((StatusCodes.UnprocessableEntity, err))
-                case Right(trigger) =>
-                  val uuid = UUID.randomUUID
-                  val ref = ctx.spawn(
-                    TriggerActor(
-                      TriggerActor.Config(compiledPackages, trigger, ledgerConfig, params.party)),
-                    uuid.toString,
-                  )
-                  triggers = triggers + (uuid -> ref)
-                  complete(uuid.toString)
+        concat(
+          // Start a new trigger given its identifier and the party it should be running as.
+          // Returns a UUID for the newly started trigger.
+          path("start") {
+            entity(as[TriggerParams]) {
+              params =>
+                Trigger.fromIdentifier(compiledPackages, params.identifier) match {
+                  case Left(err) =>
+                    complete((StatusCodes.UnprocessableEntity, err))
+                  case Right(trigger) =>
+                    val uuid = UUID.randomUUID
+                    val ref = ctx.spawn(
+                      TriggerActor(
+                        TriggerActor.Config(compiledPackages, trigger, ledgerConfig, params.party)),
+                      uuid.toString,
+                    )
+                    triggers = triggers + (uuid -> ref)
+                    complete(uuid.toString)
 
-              }
+                }
+            }
+          },
+          // upload a DAR as a multi-part form request with a single field called
+          // "dar".
+          path("upload_dar") {
+//            extractExecutionContext { implicit ec =>
+            fileUpload("dar") {
+              case (metadata: FileInfo, byteSource: Source[ByteString, Any]) =>
+                val byteStringF: Future[ByteString] = byteSource.runFold(ByteString(""))(_ ++ _)
+                onSuccess(byteStringF) {
+                  byteString =>
+                    val inputStream = new ByteArrayInputStream(byteString.toArray)
+                    DarReader()
+                      .readArchive("package-upload", new ZipInputStream(inputStream)) match {
+                      case Failure(err) => complete((StatusCodes.UnprocessableEntity, err))
+                      case Success(encodedDar) =>
+                        try {
+                          val dar = encodedDar.map {
+                            case (pkgId, payload) => Decode.readArchivePayload(pkgId, payload)
+                          }
+                          addDar(compiledPackages, dar)
+                          complete(s"DAR uploaded, main package id: ${dar.main._1}")
+                        } catch {
+                          case e: ParseError => complete((StatusCodes.UnprocessableEntity, e))
+                        }
+                    }
+                }
+              // foreach { byteString =>
+              //   println("GOT REQUEST")
```

Comment:
Debug left-over?

Reply:
Good catch, I clearly forgot the cleanup pass for this PR. Sorry :slightly_smiling_face: Should all be fixed now.

Reply:
No worries, thanks for fixing these!